tests/ui/lint/rfc-2383-lint-reason/expect_unused_inside_impl_block.rs

```diff
@@ -0,0 +1,25 @@
+// check-pass
```

Comment:
Could you make this an incremental test too? To check that all sessions emit the same diagnostic.

Reply:
The rustc dev guide mentions an "incremental" mode where the compiler is run multiple times in order to confirm that incremental compilation works.

When I grep for "incremental" in `tests/ui` I only see the `// incremental` (i.e. `-C incremental`) header being used, but I don't see anything that looks like its enabling this "incremental mode" (I can't quite tell how to enable that).

My understanding is that the `// incremental` header runs the test once, but your recommendation is that we should run the test once with a clean incremental
cache dir and once with the populated cache.

So, my question is, how do I make sure that the test runs once with an empty incremental cache and then again with the now populated cache?

---

https://rustc-dev-guide.rust-lang.org/tests/headers.html?highlight=incremental%20mode#miscellaneous-headers

> incremental runs the test with the -C incremental flag and an empty incremental directory. This should be avoided when possible; you should use an incremental mode test instead. Incremental mode tests support running the compiler multiple times and verifying that it can load the generated incremental cache. This flag is for specialized circumstances, like checking the interaction of codegen unit partitioning with generating an incremental cache.

Reply:
Incremental tests are in tests/incremental . There are several examples in there.